modules/api/src/main/java/org/apache/ignite/compute/JobDescriptor.java

```diff
@@ -100,12 +104,16 @@ public static <T, R> Builder<T, R> builder(Class<? extends ComputeJob<T, R>> job
         return new Builder<>(jobClass.getName());
     }
 
+    public @Nullable Marshaller<T, byte[]> argumentMarshaller() {
+        return argumentMarshaller;
+    }
+
     public @Nullable Marshaller<R, byte[]> resultMarshaller() {
         return resultMarshaller;
     }
 
-    public @Nullable Marshaller<T, byte[]> argumentMarshaller() {
-        return argumentMarshaller;
+    public @Nullable Class<?> resultClass() {
```

Comment:
So we can infer job argument type from the overridden method, but we can't do it for the result because of type erasure, and we have to ask the user to provide the class explicitly - is my understanding correct?